addons/project/models/project.py

```diff
@@ -2720,23 +2720,35 @@ def _get_default_color(self):
         ('name_uniq', 'unique (name)', "A tag with the same name already exists."),
     ]
 
-    def _get_project_tags_domain(self, domain, project_id):
-        tag_ids = list(self.with_user(SUPERUSER_ID)._search(
-            ['|', ('task_ids.project_id', '=', project_id), ('project_ids', 'in', project_id)]))
-        return expression.AND([domain, [('id', 'in', tag_ids)]])
```

Comment:
In stable we cannot remove a method, but we could deprecated it. 🙂

```py
    def _get_project_tags_domain(self, domain, project_id):
        # TODO: Remove me in master
        tag_ids = list(self.with_user(SUPERUSER_ID)._search(
            ['|', ('task_ids.project_id', '=', project_id), ('project_ids', 'in', project_id)]))
        return expression.AND([domain, [('id', 'in', tag_ids)]])
```

or we could remove the logic inside that method to directly return the domain given in parameter.

```py
    def _get_project_tags_domain(self, domain, project_id):
        # TODO: Remove me in master
        return domain
```
In that case, we completely deprecate the method. 🙂

Reply:
> In stable we cannot remove a method, but we could deprecated it. slightly_smiling_face
> 
> ```python
>     def _get_project_tags_domain(self, domain, project_id):
>         # TODO: Remove me in master
>         tag_ids = list(self.with_user(SUPERUSER_ID)._search(
>             ['|', ('task_ids.project_id', '=', project_id), ('project_ids', 'in', project_id)]))
>         return expression.AND([domain, [('id', 'in', tag_ids)]])
> ```
> 
> or we could remove the logic inside that method to directly return the domain given in parameter.
> 
> ```python
>     def _get_project_tags_domain(self, domain, project_id):
>         # TODO: Remove me in master
>         return domain
> ```
> 
> In that case, we completely deprecate the method. slightly_smiling_face

Thank you for your review, I've updated the PR ;)